packages/text-annotator/vite.config.ts

```diff
@@ -0,0 +1,36 @@
+import { defineConfig } from 'vite';
+import dts from 'vite-plugin-dts';
+import { externalizeDeps } from 'vite-plugin-externalize-deps';
+
+export default defineConfig({
+  plugins: [
+    dts({ insertTypesEntry: true, entryRoot: '.' }),
+    externalizeDeps(),
+  ],
+  server: {
+    open: '/test/index.html'
+  },
+  build: {
+    minify: false,
+    sourcemap: true,
+    lib: {
+      entry: './src/index.ts',
+      name: 'RecogitoJS',
+      formats: ['es', 'umd'],
+      fileName: (format) => `text-annotator.${format}.js`
+    },
+    rollupOptions: {
+      output: {
+        assetFileNames: 'text-annotator.[ext]',
+        globals: {
+          'colord': 'Colord',
+          'uuid': 'UUID',
+          'dequal/lite': 'DequalLite',
+          '@annotorious/core': 'AnnotoriousCore',
+          'rbush': "RBush",
+          'hotkeys-js': 'HotkeysJs',
+        }
```

Comment:
The globals are needed to bundle the UMD package using predictable names for the external packages. Otherwise, they will be guessed by Vite:
![image](https://github.com/user-attachments/assets/828c53c3-e558-4902-9c5e-b4c9368155f1)